Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp

```diff
@@ -1378,10 +1378,47 @@ void MediaPlayerPrivateGStreamerBase::dispatchDecryptionStructure(GUniquePtr<Gst
     GST_TRACE("emitted decryption structure on pipeline, event handled %s", boolForPrinting(eventHandled));
 }
 
-void MediaPlayerPrivateGStreamerBase::reportWaitingForKey()
+void MediaPlayerPrivateGStreamerBase::setWaitingForKey(bool waitingForKey)
 {
-    GST_TRACE("waiting for key");
-    m_player->waitingForKey();
+    // We bail out if values did not change or if we are requested to not wait anymore but there are still waiting decryptors.
+    GST_TRACE("waitingForKey %s, m_waitingForKey %s", boolForPrinting(waitingForKey), boolForPrinting(m_waitingForKey));
+    if (waitingForKey == m_waitingForKey || (!waitingForKey && this->waitingForKey()))
+        return;
+
+    m_waitingForKey = waitingForKey;
+    GST_DEBUG("waiting for key changed %s", boolForPrinting(m_waitingForKey));
+    m_player->waitingForKeyChanged();
+}
+
+bool MediaPlayerPrivateGStreamerBase::waitingForKey() const
+{
+    bool result { false };
+    if (m_pipeline) {
```

Comment:
This looks like quite a slow process, given that the waiting for key status and change in a very short time in practice, my gut feel is by the time this query has been answered, the duration of us actually waiting for a key will be missed. Really need test cases for this.

Reply:
It isn't. I run several test runs and gathered to each test run how long all calls to this function take (the sum of the time it takes calling this function) and the longest case is 8,6ms so a quite acceptable result IMHO for the proper compliance with the spec.

Reply:
And this times are measured on the Rpi2

Reply:
That's great :)